internal/pkg/agent/transpiler/vars.go

```diff
@@ -276,3 +294,15 @@ func varPrefixMatched(val string, key string) bool {
 	s := strings.SplitN(val, ".", 2)
 	return s[0] == key
 }
+
+func maybeAddDefaultProvider(val string, defaultProvider string) string {
+	if defaultProvider == "" {
+		return val
+	}
+	if strings.Contains(val, ".") {
+		// already has a provider in the variable name
+		return val
+	}
```

Comment:
These two condition can be collapsed and maybe a constant can be introduced (if it doesn't exist already) for the `"."` separator

Reply:
I will merge these.